src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -351,29 +361,78 @@ public static bool IsOnlyWrittenTo(this ExpressionSyntax expression)
                         return true;
                     }
                 }
+
+                if (IsExpressionOfArgumentInDeconstruction(expression))
+                {
+                    return true;
+                }
             }
 
             return false;
         }
 
-        public static bool IsAttributeNamedArgumentIdentifier(this ExpressionSyntax expression)
```

Comment:
i moved this function down as it was accidentally added between IsWrittenTo and IsOnlyWrittenTo.